packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -0,0 +1,74 @@
+import synchronized from 'decorator-synchronized'
+import { noSuchObject } from 'xo-common/api-errors'
+
+import Collection from '../collection/redis'
+import patch from '../patch'
+
+const sync = synchronized.withKey((arg1, arg2) => arg1.address ?? arg2.address)
```

Comment:
I don't like this dynamic key function, unfortunately I don't think we have a choice…

Reply:
I think we can do better: add a test on the first argument and treat each case separately.

Also handle the case where `address` is undefined for `updateProxy`.